numpy/core/arrayprint.py

```diff
@@ -272,22 +272,24 @@ def get_printoptions():
     """
     return _format_options.copy()
 
-def _leading_trailing(a):
+
+def _leading_trailing(a, index=()):
+    """
+    Keep only the N-D corners (leading and trailing edges) of an array
+    """
     edgeitems =  _format_options['edgeitems']
```

Comment:
This is a bug (it ignored the kwarg to `array2string`), but I want to leave that to another PR

Reply:
probably good to put up a separate issue for it (if you haven't done so already) or link to a PR. (I'm only arguing from knowing my state of memory -- I'd forget...)

Reply:
#10136